src/net.cpp

```diff
@@ -2916,6 +2922,19 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& address) const
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
 
+void CConnman::ASMapHealthCheck()
+{
+    const std::vector<CAddress> v4_addrs{GetAddresses(/*max_addresses=*/ 0, /*max_pct=*/ 0, Network::NET_IPV4)};
```

Comment:
Just noting that this means that addresses for which IsTerrible() is true (e.g. not seen in 30 days) will be excluded from the statistics. These addresses are not treated any different when selecting peers for new automatic connections.

Reply:
Ah, hm, that doesn't sound ideal to me. Do you know what the rationale is for excluding them from `GetAddr()` in all cases? I don't see that mentioned in `GetAddr()` header file doc comment and `GetAddr_()` just says "Filter for quality". Git blaming my way back to the original point it was added seems tedious at first look due to the many refactors on this code...

Reply:
Mostly historical reasons, `GetAddr()` was initially only used to answer `GETADDR` messages, and only later used for reporting purposes such as `getnodeaddresses`. #27511 has some more discussion on this. I don't think it is strictly necessary to do anything about it here, but one way to deal with it would be to just add a bool to  `GetAddr()`  that lets callers pick whether to filter out `Terrible` addresses or not, and use it to include all addresses in the ASMap check.

Reply:
I have implemented this per our offline conversation. Keeping this conversation unresolved though so others can chime in if they have a different opinion on this topic.